src/pages/routes.test.ts

```diff
@@ -0,0 +1,44 @@
+import fs from 'fs'
+import parser from 'xml2json'
+
+import { routes } from './RouteDefinitions'
+
+describe('Routes', () => {
+  it('sitemap URLs should exist as Router paths', async () => {
+    const pathNames: string[] = routes.map((routeDef) => routeDef.path)
+    const result = await new Promise<boolean>((resolve) => {
+      fs.readFile('./public/sitemap.xml', 'utf8', (err, data) => {
+        try {
+          const sitemap = parser.toJson(data)
+          const sitemapPaths = JSON.parse(sitemap).urlset.url.map((url: any) => new URL(url.loc).pathname)
+
+          sitemapPaths.forEach((path: string) => {
+            if (!pathNames.includes(path)) {
+              console.log(`${path} is missing from Routes`)
+              resolve(false)
+            }
+          })
+
+          resolve(true)
+        } catch {
+          resolve(false)
+        }
+      })
+    })
+    expect(result).toBeTruthy()
+  })
+
+  /**
+   * If you are updating the sitemap or app routes, consider if you need to make a
+   * corresponding update in the other file.
+   */
+
+  it('sitemap should match snapshot', () => {
+    const result = fs.readFileSync('./public/sitemap.xml', 'utf8')
+    expect(result).toMatchSnapshot()
+  })
+
+  it('router definition should match snapshot', () => {
+    expect(routes).toMatchSnapshot()
+  })
```

Comment:
Can a sitemap be generated from the JSON in the RouteDefinitions file, or must it be done manually? It'd be nice if you could write a short utility script to do this for us, especially as it seems like the timestamps will go out of date very quickly.

Reply:
i'm board with this idea in theory, but it's hard to share code between the app .tsx files and the scripts. i think the only way to do this would be to have a standalone, static JSON file with just the path names. we could use the file in both the script and the `RouteDefinitions.tsx` app code. The problem with this approach is just verbosity and having to update multiple files to add a new path - the JSON file and the App code which turns the pathname into a React element. i'm not sure if that extra friction is worth it, vs just accepting some latency in updating the sitemap. what do you think?

Reply:
oh and i added a script which keeps the timestamps up to date